packages/sol-compiler/src/solc_wrapper_v06.ts

```diff
@@ -0,0 +1,25 @@
+import { CompilerOptions, StandardOutput } from 'ethereum-types';
+import solc = require('solc');
+
+import { compileSolcJSAsync, getSolcJSAsync } from './utils/compiler';
+
+import { SolcWrapperV05 } from './solc_wrapper_v05';
+
+// 0.6.x has a `compile()` function in lieu of `compileStandardWrapper`.
+type SolcV06 = solc.SolcInstance & { compile(input: string): string };
```

Comment:
Should this be a union type rather than an intersection type? It would seem that this would remove all type information since `solc.SolcInstance` doesn't actually have a `compile` function with the same function signature as `compile(input: string): string`.

Source:
```
    export interface SolcInstance {
        compile(
            sources: InputSources,
            optimizerEnabled: number,
            findImports: (importPath: string) => ImportContents,
        ): CompilationResult;
        compileStandardWrapper(input: string, findImports?: (importPath: string) => ImportContents): string;
    }
```

Please let me know if I'm missing something here. If I am not, it may be necessary to add the new `compile` function to `typescript-typings`

Reply:
It's my understanding that intersection types _are_ unions. Union types are either-or. idk why TS picked those horrible names.  Feels weird to add the new compile method to `SolcInstance` because that's the 0.4/0.5 interface in my mind.

Reply:
Yeah that is a *horrible* name. I can't believe they chose that.

Carry on...